### DISCUSSION TITLE
How do I change the value of "Size" in Camera3D resources using mouse wheel?

### DISCUSSION THREAD

**Discussion by davidmctravord555** [ramokz/phantom-camera#387] [Question / Help Wanted] [answered]
So I'm making a top down 2.5D game, and I use PCam3D as the camera. I use orthogonal projection, so in order to zoom in or out, it used the "Size" parameter. I already assign key for zoom_in and zoom_out, and assign them to mousewheel, so my question is how do I control the value of "Size" using the zoom_in and zoom_out key? Also how do I make an export parameter for the increment too?

**Comment by ramokz** [comment]
Should be able to set the size using the property. E.g. `pcam.size = value`. In your case, you would just need to increment or decrement the size value there.

> Also how do I make an export parameter for the increment too?

Not entirely sure what you mean by this. What are you looking to achieve?

**Reply by davidmctravord555** [reply]
Okay so first, I will provide context more, and some re-asking question because I change few things. I'm making a 3D top down game similar to Cassette Beast, and at first I'm using orthogonal projection of the Camera3D in order to achieve flat 2D look, however the clipping issue is hard to fix, therefore I decided to use normal perspective projection, so the game will looks like Cassette Beast or Cult of the Lamb.

I'm setting up the PCam3D as follow mode: framed and follow target: player_camera_target (I put separate target as the child of player because player root is in the leg), and I notice that by changing the value of follow_distance, the camera moves either closer or farther from the player (in my previous case where I use orthogonal projection, I change the value of camera size parameter).

So I set up this code after the setter and getter function of follow_distance:

![image](https://github.com/user-attachments/assets/3e70e94a-c814-48f7-b14d-792f81113f97)

And what I noticed is, the camera is indeed moving with the action key pressed, but only during the initial "zooming in" phase after launching the game. After that, it locks to whatever the value being set during the end of that phase. From the debugger, I noticed that the value is indeed still changing, just not happening in game since it's being locked

![image](https://github.com/user-attachments/assets/f70184cb-ba29-4aae-90f1-bd3e184e2d2d)

So my new question is, how do I remove the lock of follow_distance value?


Also what I mean by increment, I was gonna say "steps", like how much one press of button is changing the value (the one I use here is 1). I already found a way to do that, but one more question I ask, do I need the getter and setter too?

**Reply by ramokz** [reply, marked as answer]
> And what I noticed is, the camera is indeed moving with the action key pressed, but only during the initial "zooming in" phase after launching the game. After that, it locks to whatever the value being set during the end of that phase. From the debugger, I noticed that the value is indeed still changing, just not happening in game since it's being locked.

Short answer, you can't at the minute due to how it's currently set up.
This would be part of the refactor task (#161) that's currently being planned, as there are some complications with adjusting the distance during runtime. E.g. how the camera should behave if you adjust the distance and in so doing would cause the `dead zone` to be touched. Updating during runtime only works with `Third Person Follow` due to its lesser complexity. 

> Also what I mean by increment, I was gonna say "steps", like how much one press of button is changing the value (the one I use here is 1). I already found a way to do that, but one more question I ask, do I need the getter and setter too?

Not too sure what you mean by needing the getter and setter? Are you referring to the getter and setter for the distance property? If so, then no, that's just an alternative way of getting and assigning the value respectively. It's there partly because it wasn't possible in an earlier release to change or get property directly, but also to give people the option to use either the property directly or a setter/getter function. Either approach has the same result.

**Reply by davidmctravord555** [reply]
Oh thanks for the answer, I guess I'll stick with third person follow then for now. The getter and setter one, I probably won't have to worry about it for now, thanks again~
